### ISSUE TITLE
can't tag ELBs

### ISSUE THREAD

**Issue by smiller171** [ansible/ansible-modules-core#439]
##### Issue Type:

Feature Idea
##### Component Name:

ec2_elb_lb module
##### Ansible Version:

1.8.1
##### Environment:

Managing EC2 from generic.
##### Summary:

Currently there is no way to tag an ELB because it does not have a resource_id, so it can't be tagged by ec2_tag.
##### Steps To Reproduce:

I think that you should either have a "tags" parameter in the ec2_elb_lb module (like instance_tags in the ec2 module) or allow for the use of the resource name in ec2_tag.
##### Expected Results:

I expect to be able to add tags to ELBs by using either a parameter in the ec2_elb_lb module or using the ec2_tag module.
##### Actual Results:

Currently you can't tag ELBs from Ansible.

**Comment by bobby** [comment]
+1, it would be nice to be able to uniformly tag all AWS resources provisioned via Ansible.

**Comment by nicolas-g** [comment]
+1 , tagging has become really important for cost reporting, administration and automation ...

**Comment by sebest** [comment]
+1 here

**Comment by evitorbarros** [comment]
+1

**Comment by ryanwalls** [comment]
+1

**Comment by k3ypad** [comment]
Email sent to ansible devel list - should have something soon:
https://groups.google.com/forum/#!topic/ansible-devel/FJZ8eOqhfOc

**Comment by eproxus** [comment]
What's the current status? Still missing boto support?

**Comment by smiller171** [comment]
@eproxus yup, still missing. I'm tagging ELBs in my playbooks using the shell module to run AWSCLI commands

**Comment by eproxus** [comment]
Ah, do you have a snippet you could share? :smile:

**Comment by smiller171** [comment]
```
- name: tag ELB
  shell: "aws --region us-east-1 elb add-tags --load-balancer-names {{elbname}} --tags Key=stack,Value={{stack}} Key=layer,Value={{layer}} Key=owner,Value={{owner}}"
```

**Comment by eproxus** [comment]
@smiller171 Awesome, thanks!

**Comment by smiller171** [comment]
no problem @eproxus. I think Ansible 2.0 is supposed to switch to Boto3, which does have tagging support for ELBs

**Comment by s0enke** [comment]
> no problem @eproxus. I think Ansible 2.0 is supposed to switch to Boto3, which does have tagging support for ELBs

@smiller171 Do you have any particular information about that? I could not find any hints that Ansible2 is going to be ported to Boto3.

**Comment by smiller171** [comment]
After much searching, I found nothing. I have suspicions about where this information came from, but don't recall for sure.

**Comment by groorj** [comment]
+1

**Comment by welex91** [comment]
I've created a PR in case someone would like to test this: https://github.com/ansible/ansible-modules-extras/pull/1454

**Comment by Constantin07** [comment]
+1

**Comment by welex91** [comment]
Another PR: https://github.com/ansible/ansible-modules-core/pull/2844 this time patching ec2_elb_lb, based on helpful feedback from @jonhadfield and @Constantin07

**Comment by nand0p** [comment]
https://github.com/ansible/ansible-modules-extras/pull/1738 boto3 mod.  returns changed only if elb tags are changed.

**Comment by nand0p** [comment]
https://github.com/ansible/ansible-modules-core/pull/3163 updates ec2_elb_lb module to support add + modify tags

**Comment by shay-li** [comment]
+1

**Comment by ericcitaire** [comment]
https://github.com/ansible/ansible-modules-extras/pull/2133 adds a new module ec2_elb_tag.

**Comment by ansibot** [comment]
@jsdalton, @rickmendes, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@jsdalton, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@jsdalton, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by smiller171** [comment]
This issue was moved to ansible/ansible#18927
